### PR TITLE
Add exitOnWait flag

### PR DESCRIPTION
**Pull request by kfox1111**
This enables the exitOnWait behavior from the commandline so most users don't need to define two different config files.

### CODE REVIEW

**Comment by kfox1111** [comment]
There should probably be a bigger conversation about having all config file options also be flags. But until then, this PR will enable a pretty good savings on configuration.

File: pkg/sidecar/sidecar_test.go [hunk 1]
```diff
@@ -194,14 +194,14 @@ func TestSidecar_RunDaemon(t *testing.T) {
 
 func TestDefaultAgentAddress(t *testing.T) {
 	log, _ := test.NewNullLogger()
-	spiffeSidecar, err := New("../../test/sidecar/config/helper.conf", log)
+	spiffeSidecar, err := New("../../test/sidecar/config/helper.conf", false, log)
```
Comment (MarcosDY) [inline pkg/sidecar/sidecar_test.go:197]: can you add a true case?

**Review by MarcosDY — COMMENTED** [review]
I agree we must provide more flags here... but this is a good example,
if we go for that approach, I think we may move the config parsing to another place, to simplify changing values from configs from flags...

File: pkg/sidecar/sidecar_test.go (hunk 1 above)
Comment (kfox1111) [reply pkg/sidecar/sidecar_test.go:197]: Done

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**